packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -0,0 +1,238 @@
+
+    /// @notice The delay time in seconds between when an upgrade is initiated and when it can be finalized.
+    ///         It can only be modified by an upgrade.
+    bytes32 public constant DELAY_SLOT = bytes32(uint256(keccak256("superchainConfig.delay")) - 1);
+
+    /// @notice The time until which the system is paused.
+    bytes32 public constant PAUSED_SLOT = bytes32(uint256(keccak256("superchainConfig.paused")) - 1);
+
+    // todo(maurelian): make this time configurable. It was just a lot easier to mock it up by hardcoding it in.
+    /// @notice The maximum time in seconds that the system can be paused for.
+    uint256 public constant maxPause = 1 weeks;
+
+    /// @notice Mapping of allowed sequencers.
+    ///         The initiator should be able to add to it instantly, but removing is subject to delay.
+    mapping(bytes32 => bool) public allowedSequencers;
+
+    /// @notice Emitted when the pause is triggered.
+    event Paused();
+
+    /// @notice Emitted when the pause is lifted.
+    event Unpaused();
+
+    /// @notice Emitted when configuration is updated.
+    /// @param updateType Type of update.
+    /// @param data       Encoded update data.
+    event ConfigUpdate(UpdateType indexed updateType, bytes data);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Constructs the SuperchainConfig contract.
+    constructor() {
+        initialize({
+            _systemOwner: address(0),
+            _initiator: address(0),
+            _vetoer: address(0),
+            _guardian: address(0),
+            _delay: 0,
+            _sequencers: new T.SequencerKeys[](0)
+        });
+    }
+
+    /// @notice Initializer.
+    ///         The resource config must be set before the require check.
+    /// @param _systemOwner Owner of the contract.
+    /// @param _initiator   Address of the initiator who may initiate an upgrade or change to critical config values.
+    /// @param _vetoer      Address of the vetoer.
+    /// @param _guardian    Address of the guardian, can pause the OptimismPortal.
+    /// @param _delay       The delay time in seconds between when an upgrade is initiated and when it can be finalized.
+    /// @param _sequencers  The initial list of allowed sequencers
+    function initialize(
+        address _systemOwner,
+        address _initiator,
+        address _vetoer,
+        address _guardian,
+        uint256 _delay,
+        T.SequencerKeys[] memory _sequencers
+    )
+        public
+        reinitializer(2)
+    {
+        _setSystemOwner(_systemOwner);
+        _setInitiator(_initiator);
+        _setVetoer(_vetoer);
+        _setGuardian(_guardian);
+        _setDelay(_delay);
+
+        for (uint256 i = 0; i < _sequencers.length; i++) {
+            bytes32 sequencerHash = Hashing.hashSequencerKeys(_sequencers[i]);
```

Comment:
Curious if you think this should call `_addSequencer` and then make `addSequencer` have the auth check

Reply:
Yes, I like that, implemented.